### PR TITLE
fix: enhance my ticket page

### LINKED ISSUE #6138: My Ticket Page: Add links and additional data
On the my ticket page the tickets are missing some data and links. Please add the following:

Show all basic data
* from to time and date
* Online event or location or both for mixed events

Add links
* Link event name to event page
* Order number of order

Compare https://eventyay.com/my-tickets

Screenshot from 2020-12-24 16-06-13

### PR DESCRIPTION
**Pull request by Soumi150**
Fixes #6138

#### Short description of what this resolves:
Screenshot (50)

#### Changes proposed in this pull request:

Show all basic data

-from to time and date
-Online event or location 

Add links

-Link event name to event page
-Order number of order

#### Checklist

- [x] I have read the [Contribution & Best practices Guide](https://blog.fossasia.org/open-source-developer-guide-and-best-practices-at-fossasia).
- [x] My branch is up-to-date with the Upstream `development` branch.
- [x] The acceptance, integration, unit tests and linter pass locally with my changes 
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have added necessary documentation (if appropriate)

### CODE REVIEW

**Review by MeghalBisht — CHANGES_REQUESTED** [review]
Please make the changes when start & end date is same for an event as described in https://github.com/fossasia/open-event-frontend/pull/5933#issuecomment-740060025

File: app/templates/components/order-card.hbs [hunk 1]
```diff
@@ -14,18 +14,31 @@
         </UiPopup>
       </div>
       <SmartOverflow @class="header">
-        {{this.order.event.name}}
+        <a href="{{this.order.event.url}}" rel="noopener noreferrer">{{this.order.event.name}}</a>
       </SmartOverflow>
       <div class="meta">
         <span class="date">
           {{general-date this.order.event.startsAt 'date-time-tz-long' tz=this.order.event.timezone}}
```
Comment (MeghalBisht) [inline app/templates/components/order-card.hbs:21]: Also show timezone only once as in #5933

**Review by MeghalBisht — CHANGES_REQUESTED** [review]

**Comment by iamareebjamal** [comment]
Use EventDate component

File: app/templates/components/order-card.hbs [hunk 2]
```diff
@@ -18,9 +18,15 @@
...
-          -
-          {{general-date this.order.event.endsAt 'date-time-tz-long' tz=this.order.event.timezone}}
+          {{#if this.order.event.isSingleDay}}
+            {{general-date this.order.event.startsAt 'date-time-long' tz=this.order.event.timezone}}
+            -
+            {{general-date this.order.event.endsAt 'time-tz-short' tz=this.order.event.timezone}}
+          {{else}}
+            {{general-date this.order.event.startsAt 'date-time-long' tz=this.order.event.timezone}}
+            -
+            {{general-date this.order.event.endsAt 'date-time-tz-long' tz=this.order.event.timezone}}
```
Comment (iamareebjamal) [inline app/templates/components/order-card.hbs:28]: Duplicate code

File: app/templates/components/order-card.hbs [hunk 3]
```diff
@@ -18,15 +18,7 @@
...
-          {{#if this.order.event.isSingleDay}}
-            {{general-date this.order.event.startsAt 'date-time-long' tz=this.order.event.timezone}}
-            -
-            {{general-date this.order.event.endsAt 'time-tz-short' tz=this.order.event.timezone}}
-          {{else}}
-            {{general-date this.order.event.startsAt 'date-time-long' tz=this.order.event.timezone}}
-            -
-            {{general-date this.order.event.endsAt 'date-time-tz-long' tz=this.order.event.timezone}}
-          {{/if}}
+          <Public::EventDateTime @event={{@order.event}}/>
```
Comment (iamareebjamal) [inline app/templates/components/order-card.hbs:21]:
```suggestion
          <Public::EventDateTime @event={{@order.event}} />
```

File: app/templates/components/order-card.hbs [hunk 4]
```diff
@@ -14,18 +14,29 @@
...
+        </p>
+      {{/if}}
+      {{#if this.order.event.shortLocationName}}
+          <span style="color: #000000AD">
+            <i class="map marker alternate icon"></i>
+          </span>
+          <SmartOverflow @class="d-inline description">
+            {{this.order.event.shortLocationName}}
+          </SmartOverflow>
+      {{/if}}
```
Comment (iamareebjamal) [inline app/templates/components/order-card.hbs:37]: Duplicate code. Extract in component and reuse

**Comment by mariobehling** [comment]
Please add a screenshot into the description in all your PRs.

File: app/components/public/event-venue.hbs [hunk 5]
```diff
@@ -0,0 +1,14 @@
+{{#if @event.online}}
```
Comment (iamareebjamal) [inline app/components/public/event-venue.hbs:1]: Use same component in event card

**Comment by Soumi150** [comment]
@mariobehling , @iamareebjamal  should I link event name to event page in event card page too ?

**Comment by iamareebjamal** [comment]
No

File: app/templates/components/event-card.hbs [hunk 6]
```diff
@@ -23,31 +23,10 @@
...
-          {{#if this.event.isSingleDay}}
-            {{general-date this.event.startsAt 'date-time-long' tz=this.event.timezone}}
-            - 
-            {{general-date this.event.endsAt 'time-tz-short' tz=this.event.timezone}}
-          {{else}}
-            {{general-date this.event.startsAt 'date-time-long' tz=this.event.timezone}}
-            -
-            {{general-date this.event.endsAt 'date-time-tz-long' tz=this.event.timezone}}
-          {{/if}}
+          <Public::EventDateTime @event={{@event}} />
```
Comment (maze-runnar) [inline app/templates/components/event-card.hbs:26]: why removed `if` ? why removed dates (correct me if it was a requirement of issue)?

Comment (Soumi150) [reply app/templates/components/event-card.hbs:26]: @maze-runnar I was told to make changes in event card  to remove code duplicacy, please see above comment .Dates and location have been replaced by components in both order-card and event-card.

**Pull request merged**